lua/telescope/actions/init.lua

```diff
@@ -250,6 +250,22 @@ actions.git_add = function(prompt_bufnr)
   os.execute('git add ' .. val)
 end
 
+actions.git_staging_toggle = function(prompt_bufnr)
+  local selection = actions.get_selected_entry(prompt_bufnr)
+
+  -- If parts of the file are staged and unstaged at the same time, stage
+  -- changes. Else toggle between staged and unstaged if the file is tracked,
+  -- and between added and untracked if the file is untracked.
+  if selection.status:sub(2) == ' ' then
+    os.execute('git restore --staged ' .. selection.value)
+  else
+    os.execute('git add ' .. selection.value)
+  end
+  actions.close(prompt_bufnr)
+  vim.cmd [[lua require('telescope.builtin').git_status()]]
+  vim.api.nvim_feedkeys('i', 'n', false)
```

Comment:
That might be the fix @tami5 and i were looking for the past week. Will this put you in insert mode after toggling? Because `vim.cmd('startinsert')` would not.

Reply:
Yes, that puts you in insert mode after toggling. The hackiness of that line made me feel a bit guilty so I'm glad it helps haha

Reply:
@cempassi nice love it, going to steal it to finish my pr :$